repository_pool.go

```diff
@@ -25,34 +25,53 @@ func NewRepository(id string, repo *git.Repository) Repository {
 	}
 }
 
-// RepositoryPool holds a pool of initialized git repositories and
+// NewRepositoryFromPath creates and initializes a new Repository structure
+// and initializes a go-git repository
+func NewRepositoryFromPath(id, path string) (Repository, error) {
+	repo, err := git.PlainOpen(path)
+	if err != nil {
+		return Repository{}, err
+	}
+
+	return NewRepository(id, repo), nil
+}
+
+// RepositoryPool holds a pool git repository paths and
 // functionality to open and iterate them.
 type RepositoryPool struct {
-	repositories []Repository
+	repositories map[string]string
+	idOrder      []string
 }
 
 // NewRepositoryPool initializes a new RepositoryPool
 func NewRepositoryPool() RepositoryPool {
-	return RepositoryPool{}
+	return RepositoryPool{
+		repositories: make(map[string]string),
```

Comment:
Is it expected for the moment that a repo has only one path?

Reply:
Yes, in the filesystem the repo should be only once.